front/pages/api/v1/me.ts

```diff
@@ -0,0 +1,58 @@
+import type {
+  UserType,
+  WithAPIErrorResponse,
+  WorkspaceType,
+} from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { withPublicAPIAuthentication } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { apiError } from "@app/logger/withlogging";
+
+export type MeResponseBody = {
+  owner: WorkspaceType;
+  user: UserType;
+};
+
+/**
+ * @ignoreswagger
+ * WIP, undocumented.
+ * TODO(EXT): Document this endpoint.
+ */
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<MeResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.getNonNullableWorkspace();
```

Comment:
This will be null won't it?

Reply:
It;s not but it's not the truth since this is the random first workspace. This endpoint should just return the UserType which contains the workspaces memberships (see https://github.com/dust-tt/dust/blob/main/front/pages/api/user/index.ts#L41-L62)